src/libraries/System.Net.Http/src/System/Net/Http/Headers/CacheControlHeaderValue.cs

```diff
@@ -278,87 +294,49 @@ public override string ToString()
 
         public override int GetHashCode()
         {
-            // Use a different bit for bool fields: bool.GetHashCode() will return 0 (false) or 1 (true). So we would
-            // end up having the same hash code for e.g. two instances where one has only noCache set and the other
-            // only noStore.
-            int result = _noCache.GetHashCode() ^ (_noStore.GetHashCode() << 1) ^ (_maxStale.GetHashCode() << 2) ^
-                (_noTransform.GetHashCode() << 3) ^ (_onlyIfCached.GetHashCode() << 4) ^
-                (_publicField.GetHashCode() << 5) ^ (_privateField.GetHashCode() << 6) ^
-                (_mustRevalidate.GetHashCode() << 7) ^ (_proxyRevalidate.GetHashCode() << 8);
-
-            // XOR the hashcode of timespan values with different numbers to make sure two instances with the same
-            // timespan set on different fields result in different hashcodes.
-            result = result ^ (_maxAge.HasValue ? _maxAge.Value.GetHashCode() ^ 1 : 0) ^
-                (_sharedMaxAge.HasValue ? _sharedMaxAge.Value.GetHashCode() ^ 2 : 0) ^
-                (_maxStaleLimit.HasValue ? _maxStaleLimit.Value.GetHashCode() ^ 4 : 0) ^
-                (_minFresh.HasValue ? _minFresh.Value.GetHashCode() ^ 8 : 0);
+            HashCode hashcode = default;
+
+            hashcode.Add(_flags);
+            hashcode.Add(_maxAge);
+            hashcode.Add(_sharedMaxAge);
+            hashcode.Add(_maxStaleLimit);
+            hashcode.Add(_minFresh);
 
             if ((_noCacheHeaders != null) && (_noCacheHeaders.Count > 0))
             {
-                foreach (var noCacheHeader in _noCacheHeaders)
+                int unorderedHash = 0;
+                foreach (string noCacheHeader in _noCacheHeaders)
                 {
-                    result ^= StringComparer.OrdinalIgnoreCase.GetHashCode(noCacheHeader);
+                    unorderedHash ^= StringComparer.OrdinalIgnoreCase.GetHashCode(noCacheHeader);
                 }
+                hashcode.Add(unorderedHash);
             }
 
             if ((_privateHeaders != null) && (_privateHeaders.Count > 0))
             {
-                foreach (var privateHeader in _privateHeaders)
+                int unorderedHash = 0;
+                foreach (string privateHeader in _privateHeaders)
                 {
-                    result ^= StringComparer.OrdinalIgnoreCase.GetHashCode(privateHeader);
+                    unorderedHash ^= StringComparer.OrdinalIgnoreCase.GetHashCode(privateHeader);
                 }
+                hashcode.Add(unorderedHash);
             }
 
-            if ((_extensions != null) && (_extensions.Count > 0))
-            {
-                foreach (var extension in _extensions)
-                {
-                    result ^= extension.GetHashCode();
-                }
-            }
+            hashcode.Add(NameValueHeaderValue.GetHashCode(_extensions));
```

Comment:
You could use HashCode.Combine as it has an overload that takes 8 values and there are 8 values here.